app/models/ems_refresh/save_inventory_helper.rb

```diff
@@ -178,21 +178,11 @@ def relation_values(association, target)
     top_level && (target == true || target.nil? || parent == target) ? :use_association : []
   end
 
-  def get_cluster(ems, cluster_hash, rp_hash, dc_hash)
-    cluster = EmsCluster.find_by(:ems_ref => cluster_hash[:ems_ref], :ems_id => ems.id)
-    if cluster.nil?
-      rp = ems.resource_pools.create!(rp_hash)
-
-      cluster = ems.clusters.create!(cluster_hash)
-
-      cluster.add_resource_pool(rp)
-      cluster.save!
-
-      dc = Datacenter.find_by(:ems_ref => dc_hash[:ems_ref], :ems_id => ems.id)
-      dc.add_cluster(cluster)
-      dc.save!
+  def disconnect_using_association(ems, target, disconnect = true)
```

Comment:
The method name makes it sound like it's going to do the disconnect.

What about:

* `calculate_disconnects`
* `determine_disconnects`
* `find_disconnects` (but that has an AR overlap)

Something like that?